lib/web/machineid_test.go

```diff
@@ -357,6 +358,194 @@ func TestEditBot(t *testing.T) {
+		BotName: botName,
+		Roles:   []string{"test-role"},
+		Traits: []*machineidv1.Trait{
+			{
+				Name:   "test-trait-1",
+				Values: []string{"value-1"},
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	response, err := pack.clt.PutJSON(ctx, fmt.Sprintf("%s/%s", endpointV2, botName), updateBotRequestV2{
+		Roles: []string{"new-new-role"},
+	})
+	require.NoError(t, err)
+
+	var bot machineidv1.Bot
+	require.NoError(t, json.Unmarshal(response.Bytes(), &bot), "invalid response received")
+	assert.Equal(t, http.StatusOK, response.Code(), "unexpected status code updating bot")
+	assert.Equal(t, botName, bot.GetMetadata().GetName())
+	assert.Equal(t, []string{"new-new-role"}, bot.GetSpec().GetRoles())
+	assert.Equal(t, []*machineidv1.Trait{
+		{
+			Name:   "test-trait-1",
+			Values: []string{"value-1"},
+		},
+	}, bot.GetSpec().Traits)
+	assert.Equal(t, int64((12*time.Hour)/time.Second), bot.GetSpec().GetMaxSessionTtl().GetSeconds())
+}
+
+func TestEditBotTraits(t *testing.T) {
+	ctx := context.Background()
+	env := newWebPack(t, 1)
+	proxy := env.proxies[0]
+	pack := proxy.authPack(t, "admin", []types.Role{services.NewPresetEditorRole()})
+	clusterName := env.server.ClusterName()
+	endpointV1 := pack.clt.Endpoint(
+		"v1",
+		"webapi",
+		"sites",
+		clusterName,
+		"machine-id",
+		"bot",
+	)
+	endpointV2 := pack.clt.Endpoint(
+		"v2",
+		"webapi",
+		"sites",
+		clusterName,
+		"machine-id",
+		"bot",
+	)
+
+	// create a bot named `test-bot-edit`
+	botName := "test-bot-edit"
+	_, err := pack.clt.PostJSON(ctx, endpointV1, CreateBotRequest{
+		BotName: botName,
+		Roles:   []string{"test-role"},
+		Traits: []*machineidv1.Trait{
+			{
+				Name:   "test-trait-1",
+				Values: []string{"value-1", "value-2"},
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	response, err := pack.clt.PutJSON(ctx, fmt.Sprintf("%s/%s", endpointV2, botName), updateBotRequestV2{
+		Traits: []updateBotRequestTrait{
+			{
+				Name:   "test-trait-1",
+				Values: []string{"value-1", "value-2", "value-3"},
+			},
+			{
+				Name:   "test-trait-2",
+				Values: []string{"value-1"},
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	var bot machineidv1.Bot
+	require.NoError(t, json.Unmarshal(response.Bytes(), &bot), "invalid response received")
+	assert.Equal(t, http.StatusOK, response.Code(), "unexpected status code updating bot")
+	assert.Equal(t, botName, bot.GetMetadata().GetName())
+	assert.Equal(t, []string{"test-role"}, bot.GetSpec().GetRoles())
+	assert.Len(t, bot.GetSpec().Traits, 2)
+	for _, trait := range bot.GetSpec().Traits {
+		// Avoid ordering race - traits are stored in a map
+		switch trait.GetName() {
+		case "test-trait-1":
+			assert.Equal(t, []string{"value-1", "value-2", "value-3"}, trait.GetValues())
+		case "test-trait-2":
+			assert.Equal(t, []string{"value-1"}, trait.GetValues())
+		}
+	}
+	assert.Equal(t, int64((12*time.Hour)/time.Second), bot.GetSpec().GetMaxSessionTtl().GetSeconds())
+}
+
+func TestEditBotMaxSessionTtl(t *testing.T) {
```

Comment:
```suggestion
func TestEditBotMaxSessionTTL(t *testing.T) {
```

Capitalize initialisms in Go.